packages/fluentui/docs/src/prototypes/table/ResponsiveTable.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { Table } from '@fluentui/react';
+import calculateBreakpoints, { Column, Breakpoint } from '../../utils/calculateBreakpoints';
+import { TableProps } from '../../../../react/src/components/Table/Table';
+
+interface ResponsiveTableProps extends TableProps {
+  label: string;
+  header: {
+    items: Column[];
+  };
+}
+
+const ResponsiveTable = (props: ResponsiveTableProps) => {
+  const {
+    rows,
+    label,
+    header: { items }
+  } = props;
+
+  const columns = items.map(({ title }) => title);
+
+  const [breakpoints, setBreakpoints] = React.useState<Breakpoint[]>();
+
+  // Generate the stylesheet once and for all
+  React.useLayoutEffect(() => {
+    const breakpoints = Array.from(calculateBreakpoints(items, { _: 16 }));
+    setBreakpoints(breakpoints);
+  }, []);
+
+  return (
+    <div id="ResponsiveTable">
+      <style>
+        {items.map((c, i) => `#ResponsiveTable .ui-table__cell:nth-child(${i + 1}) { flex: ${c.ratio}; }`).join('\n')}
+        {breakpoints &&
+          breakpoints.map(
+            breakpoint => `
+            /* Viewport ${breakpoint.viewport} | Table ${breakpoint.table} | ${breakpoint.columns
+              .map((column, index) => `${columns[index]} ${column.state}${column.state === 'visible' ? ` (${column.size}px)` : ''}`)
+              .join(' | ')} */
+              @media (max-width: ${breakpoint.viewport}px) {
+              ${breakpoint.columns
+                .map(
+                  (column, index) =>
+                    `#ResponsiveTable .ui-table__cell:nth-child(${index + 1}) { display: ${
```

Comment:
Please could you create a variable for the wrapping <div> `id` and replace all instances of `ResponsiveTable` with that variable, so it's more clear what styles are bound by this wrapping <div>

Reply:
done